backend/service/k8s/cache.go

```diff
@@ -0,0 +1,126 @@
+package k8s
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv1 "k8s.io/api/autoscaling/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+
+	topologyv1 "github.com/lyft/clutch/backend/api/topology/v1"
+)
+
+func (s *svc) CacheEnabled() bool {
+	return true
+}
+
+func (s *svc) GetTopologyObjectChannel(ctx context.Context) chan *topologyv1.UpdateCacheRequest {
```

Comment:
what happens if this function is called twice? do we need to protect against that?

Reply:
Yeah we need to protect against this in some way